app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt

```diff
@@ -4,30 +4,121 @@ import android.content.Intent
 import android.os.Bundle
 import android.view.MenuItem
 import android.view.View
+import android.widget.ArrayAdapter
+import android.widget.ListView
+import android.widget.SearchView
 import androidx.appcompat.app.ActionBarDrawerToggle
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.GravityCompat
 import androidx.drawerlayout.widget.DrawerLayout
-import com.google.android.material.navigation.NavigationView
-
 import com.github.h3lp3rs.h3lp.presentation.PresArrivalActivity
 import com.github.h3lp3rs.h3lp.signin.ORIGIN
+import com.google.android.material.navigation.NavigationView
+import com.google.android.material.snackbar.Snackbar
 
 const val EXTRA_NEARBY_UTILITIES = "nearby_utilities"
 
+//Elements of the list view
+const val PROFILE = "Profile"
+const val CPR_RATE = "CPR rate"
+const val TUTORIAL = "Tutorial"
+
 /**
  * Main page of the app
  */
 class MainPageActivity : AppCompatActivity() {
     private lateinit var toggle: ActionBarDrawerToggle
 
+    private lateinit var searchView: SearchView
+    private lateinit var listView: ListView
+
+    //List of searchable elements
+    private var searchBarElements: ArrayList<String> = ArrayList()
+
+    //Adapter for the list view
+    lateinit var adapter: ArrayAdapter<*>
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
 
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main_page)
 
+        setUpDrawerLayout()
+
+        searchView = findViewById(R.id.searchBar)
+        listView = findViewById(R.id.listView)
+
+        //List view is invisible in the beginning.
+        listView.visibility = View.GONE
+
+        setUpListViewItems()
+
+        adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, searchBarElements)
+        listView.adapter = adapter
+
+        setUpSearchView()
+
+    }
+
+    /**
+     * Add elements to the list view
+     */
+    private fun setUpListViewItems() {
+        searchBarElements.add(PROFILE)
+        searchBarElements.add(CPR_RATE)
+        searchBarElements.add(TUTORIAL)
+    }
+
+    /**
+     * Handles the search feature on the home page. The list view is only visible when characters
+     * are entered in the search field. On text submit, an appropriate message is displayed and a
+     * new activity is launched.
+     */
+    private fun setUpSearchView() {
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+
+            override fun onQueryTextSubmit(query: String): Boolean {
+                if (searchBarElements.contains(query)) {
+                    adapter.filter.filter(query)
+                } else {
+                    displayErrorMessage()
+                }
+                return false
+            }
+
+            override fun onQueryTextChange(newText: String): Boolean {
+                if (newText.isEmpty()) {
+                    //When the text field of the search bar is empty, the list is hidden
```

Comment:
Purely aestetic but it makes comments easier to read :)

```suggestion
                    // When the text field of the search bar is empty, the list is hidden
```